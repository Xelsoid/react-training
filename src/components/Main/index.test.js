import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import MOCKED_DATA from '@mockedData/MOCKED_DATA';
import { BrowserRouter } from 'react-router-dom';
import ComponentExample from './index';

const mockStore = configureStore([]);

describe('Main test', () => {
  let store;

  it('renders gallery correctly', () => {
    store = mockStore({
      moviesData: MOCKED_DATA,
    });
    const component = mount(
      <Provider store={store}>
        <BrowserRouter>
          <ComponentExample />
        </BrowserRouter>
      </Provider>,
    );
    expect(component).toMatchSnapshot();
  });

  it('renders not found correctly', () => {
    store = mockStore({
      moviesData: {},
    });
    const component = mount(
      <Provider store={store}>
        <BrowserRouter>
          <ComponentExample />
        </BrowserRouter>
      </Provider>,
    );
    expect(component).toMatchSnapshot();
  });
});
