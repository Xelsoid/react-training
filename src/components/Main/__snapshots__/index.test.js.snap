// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main test renders gallery correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Main>
        <div
          className="main-wrapper"
        >
          <main />
        </div>
      </Main>
    </Router>
  </BrowserRouter>
</Provider>
`;

exports[`Main test renders not found correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Main>
        <div
          className="main-wrapper"
        >
          <main />
        </div>
      </Main>
    </Router>
  </BrowserRouter>
</Provider>
`;
